docs/integrations/test-explorer.md

```diff
@@ -0,0 +1,72 @@
+---
+id: test-explorer
+title: Test Explorer
+---
+
+Test Explorer is being under active development, this page explain how it works under the hood.
+
+## Test Discovery
+
+Test Explorer provides two ways to interact with it:
+
+- first one is based on notifications being sent to the client and doesn't require from client to start any actions. All necessary updates are being immediately sent to it.
+
+![](../diagrams/out/test-explorer-server-notifications.png)
+
+- client can also manually ask for test suites from the whole workspace, or for test cases for a given file.
+
+![](../diagrams/out/test-explorer-client-queries.png)
+
+## How it works under the hood
+
+Test Explorer uses BSP's [debug request](https://build-server-protocol.github.io/docs/specification.html#debug-request) to run and debug test classes. TE reuses already available debug support in Metals and build servers such as Bloop or SBT.
```

Comment:
```suggestion
Test Explorer uses BSP's [debug request](https://build-server-protocol.github.io/docs/specification.html#debug-request) to run and debug test classes. TE reuses already available debug support in Metals and build servers such as Bloop or SBT.
```
I think this is the first place you use TE so you should do Test Explorer (TE) and then use TE form there on out, or just use Test Explorer.